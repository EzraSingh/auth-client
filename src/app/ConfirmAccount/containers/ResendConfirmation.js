import React, { Component } from 'react'
import { Redirect } from 'react-router-dom'
import { toast } from 'react-toastify'
import api from 'api'
import EmailForm from '../components/EmailForm'

class ResendConfirmation extends Component{
    state = { complete : false }
    sendRecoveryLink = event => {
        event.preventDefault()
        const data = new FormData(event.target)
        api.patch("/auth", { email : data.get("email") })
        .then((res) => {
            if(res.status === 200){
                toast.success("New confirmation email sent")
                this.setState({ complete: true })
            }
        })
        .catch((err) => {
            const { message } = err.response.data
            if(err.response.status === 401){              
                toast.warn("No user with that email")
            }
            if(err.response.status === 400){
                toast.error(message)
            }
        })
    }
    render(){
        if(this.state.complete) return <Redirect to="/"/> 
        return <EmailForm handleSubmit={this.sendRecoveryLink}/>
    }
}

export default ResendConfirmation