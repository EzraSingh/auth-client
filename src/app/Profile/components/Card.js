import React from 'react'
import { isEmpty } from 'lodash'
import { Link } from 'react-router-dom'

const Card = ({ email, profile }) => {
    const avatar = profile.avatar || process.env.REACT_APP_DEFAULT_AVATAR 
    return(
        <article className="card">
            <div className="avatar">
                <figure>
                    <img src={avatar} alt="avatar"/>
                </figure>
            </div>
            <div className="details">
                <h1>{profile.name}</h1>
                <h2>{email}</h2>
                <div className="options">
                    <Link className="button button-outline" to="/profile/update">
                        Update Profile
                    </Link>
                </div>
            </div>
        </article>
    )
}

export default Card