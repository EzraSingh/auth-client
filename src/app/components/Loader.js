import React from 'react'

const Loader = () => {
    return <div className='sk-rotating-plane'/>
}

export default Loader