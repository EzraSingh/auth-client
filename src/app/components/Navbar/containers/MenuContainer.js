import React, { Component } from 'react'
import { connect } from 'react-redux'
import Menu from '../components/Menu'

/**
 * Route template for registering routes within Navbar
 * @param {} name - Nav item title 
 * @param {} href - Nav link
 * @param {} auth - display only for logged in users
 * @param {} hideOnAuth - display only for anonymous users
 */
function Route(name, href, auth, hideOnAuth){
    this.name = name
    this.href = href
    this.auth = auth
    this.hideOnAuth = hideOnAuth
}

class MenuContainer extends Component{
    static routes = [
        new Route('Profile', '/profile', { auth: true }),
        new Route('Logout', '/logout', { auth: true })
    ]
    /** Filter routes that require authentication */
    generateRoutes(){
        if(!this.props.isLoggedIn){
            // Filter routes that do not require auth
            return MenuContainer.routes.filter((route) => !route.auth)
        }
        else{
            // Filter routes that require auth and are not flagged to hide on auth
            return MenuContainer.routes.filter((route) => route.auth && !route.hideOnAuth)
        }
    }
    render(){
        return <Menu routes={this.generateRoutes()} {...this.props}/>
    }
}

const mapStateToProps = state => {
    return { 
        ...state.auth, 
        profile: {
            avatar: state.user.profile.avatar,
            name: state.user.profile.name
        }
    }
}

export { MenuContainer }
export default connect(mapStateToProps)(MenuContainer)