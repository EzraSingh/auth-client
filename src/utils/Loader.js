import React from 'react'

const Loader = () => {
    return(
        <em>Loading...</em>
    )
}

export default Loader