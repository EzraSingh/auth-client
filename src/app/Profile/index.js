import React from 'react'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router-dom'
import { withNav } from 'app/components/Navbar'
import Menu from './containers/Menu'
import Home from './containers/Home'
import Update from './containers/Update'
import Footer from 'app/components/Footer'

const View = ({ action }) => {
    switch(action){
        case 'update':
            return <Update/>
        case 'home':
            return <Home/>
        default:
            return <Redirect to="/profile/home"/>
    }
}

const Profile = ({ match }) => {
    return(
        <section id="profile" className="view">
            <Menu/>
            <div className="container">
                <View {...match.params}/>
            </div>
            <Footer/>
        </section>
    )
}

View.propTypes = { action: PropTypes.string }

Profile.propTypes = {
    match: PropTypes.shape({
        params: PropTypes.object.isRequired
    }).isRequired
}

export default withNav(Profile)