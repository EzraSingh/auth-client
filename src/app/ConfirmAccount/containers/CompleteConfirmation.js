import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router-dom'
import { toast } from 'react-toastify'
import api from 'api'
import Loader from 'app/components/Loader'

class CompleteConfirmation extends Component{
    state = { isLoading: true, complete: false }
    static propTypes = {
        token: PropTypes.string.isRequired
    }

    componentDidMount(){
        api.post("/validate/confirmation", this.props)
        .then((res) => {
            if(res.status === 200){
                toast.success('Registration complete')
                this.setState({ isLoading: false, complete: true })
            }
        })
        .catch((err) => {
            if(err.status === 400){
                this.setState({ isLoading: false, complete: false })
            }
        })
    }

    render(){
        const { isLoading, complete } = this.state
        if(isLoading) return <Loader/>
        if(!isLoading && complete) return <Redirect to="/"/>
        return(
            <section className="container">
                <h3>Invalid Token</h3>
                <p>The confirmation token provided is either <i>invalid</i> or <i>expired</i></p>
            </section>
        )
    }
}

export default CompleteConfirmation