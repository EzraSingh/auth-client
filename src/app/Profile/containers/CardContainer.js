import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Loader from 'utils/Loader'
import Card from '../components/Card'
import { connect } from 'react-redux'
import { profile } from 'actions/user'

class CardContainer extends Component{
    static propTypes = {
        isLoading: PropTypes.bool.isRequired,
        email: PropTypes.string,
        profile: PropTypes.object,
        loadProfile: PropTypes.func.isRequired
    }
    componentWillMount(){
        this.props.loadProfile()
    }
    /** Check if profile state has updated */
    shouldComponentUpdate(nextProps){
        return !!nextProps.email ^ this.props.email
    }
    render(){
        const { isLoading, email, profile } = this.props
        return isLoading ? <Loader/> : <Card {...{email, profile}}/>
    }
}

const mapStateToProps = state => {
    return { ...state.user, isLoading: !state.user.email }
}

const mapDispatchToProps = dispatch => {
    return { loadProfile: () => dispatch(profile()) }
}

export { CardContainer }
export default connect(mapStateToProps, mapDispatchToProps)(CardContainer)