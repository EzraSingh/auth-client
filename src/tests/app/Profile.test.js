import React from 'react'
import { shallow } from 'enzyme'
import Loader from 'utils/Loader'
import Profile from 'app/Profile'
import { CardContainer } from 'app/Profile/containers/CardContainer'
import Card from 'app/Profile/components/Card'

describe('profile view', () => {
    describe('<Profile/>', () => {
        it('should render', () => {
            const component = shallow(<Profile/>)
            expect(component).toMatchSnapshot()
        })
    })
    describe('<CardContainer/>', () => {
        const props = { 
            isLoading: false,
            email: 'test@user.com',
            profile: {},
            loadProfile: jest.fn()
        }
        it('should load profile on mount', () => {
            shallow(<CardContainer {...props}/>)
            expect(props.loadProfile).toHaveBeenCalledTimes(1)
        })
    })
    describe('<Card/>', () => {
        const props = { 
            email: 'test@user.com',
            profile: { name: 'John Doe', avatar: 'img.png' }
        }
        it('should render user profile', () => {
            const component = shallow(<Card {...props}/>)
            const avatar = component.find('img')
            expect(avatar.props().src).toEqual(props.profile.avatar)
            const name = component.find('h1')
            expect(name.text()).toEqual(props.profile.name)
            const email = component.find('h2')
            expect(email.text()).toEqual(props.email)
        })
    })
})