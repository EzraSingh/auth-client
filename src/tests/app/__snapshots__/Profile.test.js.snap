// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile view <Profile/> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Navbar) />,
        <div
          className="flex v-center h-center"
          id="card-container"
        >
          <Connect(CardContainer) />
        </div>,
      ],
      "className": "profile overlay",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(CardContainer) />,
          "className": "flex v-center h-center",
          "id": "card-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Navbar) />,
          <div
            className="flex v-center h-center"
            id="card-container"
          >
            <Connect(CardContainer) />
          </div>,
        ],
        "className": "profile overlay",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(CardContainer) />,
            "className": "flex v-center h-center",
            "id": "card-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
